src/pages/index.js

```diff
@@ -89,31 +134,105 @@ function handleImageClick(data) {
 /*Event Handler*/
 
 function handleProfileEditSubmit(inputValue) {
+  api.updateProfile({
+    name: inputValue.title,
+    description: inputValue.description
+  })
+  .then(updatedUserData => {
     userInfo.setUserInfo({
-        name: inputValue.title,
-        description: inputValue.description,
+      name: updatedUserData.name,
+      about: updatedUserData.about,
     });
     popupWithEditProfileForm.close();
+  })
+  .catch(err => {
+    console.error(err);
+  });
 }
 
 function handleAddCardFormSubmit(inputValue) {
     const cardData = {
         name: inputValue.title,
         link: inputValue.url,
     };
-    renderCard(cardData);
     popupWithAddCardForm.close();
     addCardForm.reset();
```

Comment:
You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.